src/nitrokey/nk3/updates.py

```diff
@@ -351,6 +396,33 @@ def _get_bootloader(self, device: TrussedBase) -> Iterator[NK3Bootloader]:
         else:
             raise self.ui.error(f"Unexpected Nitrokey 3 device: {device}")
 
+    def _check_migrations(
+        self,
+        variant: Union[Variant, AdminAppVariant],
+        current_version: Optional[Version],
+        new_version: Version,
+        status: Optional[Status],
+    ) -> frozenset["_Migration"]:
+        try:
+            migrations = _Migration.get(
+                variant=variant, current=current_version, new=new_version
+            )
+        except ValueError as e:
+            raise self.ui.error(str(e))
+
+        txt = _get_extra_information(migrations)
+        self.ui.confirm_extra_information(txt)
+
+        if _Migration.IFS_MIGRATION_V2 in migrations:
+            if status and status.ifs_blocks is not None and status.ifs_blocks < 5:
+                raise self.ui.error(
+                    "Not enough space on the internal filesystem to perform the firmware "
+                    "update. See the release notes for more information: "
+                    "https://github.com/Nitrokey/nitrokey-3-firmware/releases/tag/v1.8.2-test.20250312"
+                )
```

Comment:
Didn't we want to have a `--force` flag?

While it's good to have the warning at 5 blocks, it can work with much less available blocks. And also if the user doesn't have any resident-keys the number of blocks should be manually ignorable.